spec/tanker/user_token_spec.rb

```diff
@@ -1,19 +1,46 @@
 # frozen_string_literal: true
 RSpec.describe Tanker::UserToken do
-  it 'has a version number' do
-    expect(Tanker::UserToken::VERSION).not_to be nil
-  end
-
-  it 'returns a valid token signed with the trustchain private key' do
-    trustchain = {
+  before do
+    @trustchain = {
       id: 'AzES0aJwDCej9bQVY9AUMZBCLdX0msEc/TJ4DOhZaQs=',
       pk: 'dOeLBpHz2IF37UQkS36sXomqEcEAjSyCsXZ7irn9UQA=',
       sk: 'cBAq6A00rRNVTHicxNHdDFuq6LNUo6gAz58oKqy9CGd054sGkfPYgXftRCRLfqxeiaoRwQCNLIKxdnuKuf1RAA=='
     }
+    @user_id = 'matz@tanker.io'
+  end
+
+  it 'has a version number' do
+    expect(Tanker::UserToken::VERSION).not_to be nil
+  end
+
+  it 'raises a TypeError if argument is not a string' do
+    args = [@trustchain[:id], @trustchain[:sk], @user_id]
+    not_a_string = [nil, {}, 1234]
+
+    3.times do |position|
```

Comment:
This is really hard to read. Why not something like `not_at_string.each`  ?

Reply:
Changed in a regular `for` loop. Not so common in Ruby but makes things clearer here.